server/src/main/java/org/elasticsearch/gateway/ReplicaShardAllocator.java

```diff
@@ -110,7 +111,7 @@ public void processExistingRecoveries(RoutingAllocation allocation) {
                             "existing allocation of replica to [" + currentNode + "] cancelled, can perform a noop recovery on ["+
                                 nodeWithHighestMatch + "]",
                             null, 0, allocation.getCurrentNanoTime(), System.currentTimeMillis(), false,
-                            UnassignedInfo.AllocationStatus.NO_ATTEMPT);
+                            UnassignedInfo.AllocationStatus.NO_ATTEMPT, Collections.emptySet());
```

Comment:
I am not sure if we should carry over the failedNodeIds and failedCount here. If we have two broken nodes with noop copies, then we can repeatedly cancel an ongoing recovery for them. Note that this is not an issue with the current implementation as we ignore the failed nodes for the next round of allocation only.

Reply:
I pushed https://github.com/elastic/elasticsearch/pull/48265/commits/89c675346b2d932bc07d4f94932766e52c57e2e6 to carry over the failedNodeIds but not the failedAllocations.